hazelcast/src/main/java/com/hazelcast/query/impl/TypeConverters.java

```diff
@@ -91,15 +91,20 @@ Comparable convertInternal(Comparable value) {
     static class SqlTimestampConverter extends BaseTypeConverter {
         @Override
         Comparable convertInternal(Comparable value) {
-            if (value instanceof Timestamp) {
-                return value;
+            if (value instanceof java.sql.Timestamp) {
+                return ((java.sql.Timestamp) value).getTime();
```

Comment:
Good catch too, it's breaking the comparision of two Timestamps. Let me fixed that.